spring-kafka-docs/src/main/antora/modules/ROOT/pages/kafka/micrometer.adoc

```diff
@@ -109,3 +109,6 @@ Starting with version 3.0.6, you can add dynamic tags to the timers and traces,
 To do so, add a custom `KafkaListenerObservationConvention` and/or `KafkaTemplateObservationConvention` to the listener container properties or `KafkaTemplate` respectively.
 The `record` property in both observation contexts contains the `ConsumerRecord` or `ProducerRecord` respectively.
 
+The sender and receiver contexts' `remoteServiceName` properties are set to the Kafka `clusterId` property; this is retrieved by a `KafkaAdmin`.
+If, for some reason - perhaps lack of admin permissions, you cannot retrieve the cluster id, starting with version 3.1, you can set a manual `clusterId` on the `KafkaAdmin` and inject it into `KafkaTemplate` s and listener containers.
+When it is `null` (default), the admin will invoke the `describeCluster` admin operation to retrieve it from the broker.
```

Comment:
Don't we need to adjust `KafkaAdmin.initialize()` logic for this setting then?
There is an uncoditiional code like this:
```
synchronized (this) {
	this.clusterId = adminClient.describeCluster().clusterId().get(this.operationTimeout,
								TimeUnit.SECONDS);
}
```

Reply:
Good catch.